nats/pkg/server/nats/nats.go

```diff
@@ -4,64 +4,56 @@ import (
 	"context"
 	"time"
 
-	natsServer "github.com/nats-io/nats-server/v2/server"
-	stanServer "github.com/nats-io/nats-streaming-server/server"
+	nserver "github.com/nats-io/nats-server/v2/server"
 )
 
 var NATSListenAndServeLoopTimer = 1 * time.Second
 
 type NATSServer struct {
-	ctx context.Context
-
-	natsOpts *natsServer.Options
-	stanOpts *stanServer.Options
-
-	server *stanServer.StanServer
+	ctx             context.Context
+	jetStreamConfig *nserver.JetStreamConfig
+	server          *nserver.Server
 }
 
-// NewNATSServer returns a new NATSServer
-func NewNATSServer(ctx context.Context, opts ...Option) (*NATSServer, error) {
+func NewNATSServer(ctx context.Context, logger nserver.Logger, opts ...Option) (*NATSServer, error) {
+	options := &nserver.Options{}
 
-	server := &NATSServer{
-		ctx:      ctx,
-		natsOpts: &stanServer.DefaultNatsServerOptions,
-		stanOpts: stanServer.GetDefaultOptions(),
+	for _, o := range opts {
+		o(options)
 	}
 
-	for _, o := range opts {
-		o(server.natsOpts, server.stanOpts)
+	server, err := nserver.NewServer(
+		options,
+	)
+	if err != nil {
+		return nil, err
 	}
 
-	return server, nil
+	server.SetLoggerV2(logger, true, true, false)
+
+	c := &nserver.JetStreamConfig{
+		StoreDir: "/tmp/ocis-jetstream", // TODO: configurable
+	}
+
+	return &NATSServer{
+		ctx:             ctx,
+		jetStreamConfig: c,
+		server:          server,
+	}, nil
 }
 
 // ListenAndServe runs the NATSServer in a blocking way until the server is shutdown or an error occurs
 func (n *NATSServer) ListenAndServe() (err error) {
-	n.server, err = stanServer.RunServerWithOpts(
-		n.stanOpts,
-		n.natsOpts,
-	)
+	// start NATS first
+	go n.server.Start()
+	// start NATS JetStream second
+	n.server.EnableJetStream(n.jetStreamConfig)
```

Comment:
Question: Do we need to wait until the nats server is up? Or will it work even while the nats server is starting?

Reply:
Since `n.server.Start()` runs in a seperate goroutine, do we need a waitgroup or something to avoid `EnableJetStream` from being called before the server has started?

Reply:
I dug around in the nats server code again. There is a nicer solution which you can see in the last commit. Nats server now starts Jetstream on it's own.